quinn-proto/src/transport_parameters.rs

```diff
@@ -300,9 +301,7 @@ impl TransportParameters {
         }
         apply_params!(write_params);
 
-        // Add a reserved parameter to keep people on their toes
-        w.write_var(31 * 5 + 27);
-        w.write_var(0);
+        write_random_grease_reserved_parameter(w, &mut rand::thread_rng());
```

Comment:
Can we pipe in the endpoint/connection rng rather than using a global one? That should help with reproducibility in testing.

Reply:
`Endpoint` and `Connection` both already have `rng` fields, which are used for randomness in logic owned by those components. `RandomConnectionIdGenerator` and the config structs are constructed by the user outside of the lifetime of an endpoint or connection, and their global randomness is intended only as a convenience for typical use.

For this change, taking a generic `Rng` parameter in `TransportParameters::new` would probably be the most graceful approach, as it's called directly from `Endpoint` where the endpoint's RNG is readily available.